### ISSUE TITLE
How can I get the log during training the Yolov5 model?

### ISSUE THREAD

**Issue by gganduu** [intel-analytics/BigDL#3457]
Pytorch Yolov5 training loss will return two parameters, one is for the log, but there is only one return in Analytics Zoo loss, how can I get the training log then?

**Comment by yangw1234** [comment]
Hi @gganduu I do not understand your question. 

Could you provide more details like code snippets?

**Comment by gganduu** [comment]
Yes, we sent the code by email already.(To wang yang and qiu xin)

**Comment by yangw1234** [comment]
Synced offline, here are the brief summary:

Feature Requests:
Automatically printing loss and other metrics of every epoch or iteration when calling `Estimator.fit`

Current workaround we provides:
fitting multiple epochs in a loop. E.g.

change 
```python
est = Estimator.from_pytorch(...)
est.fit(data, epochs=epochs)
```
to 
```python
est = Estimator.from_pytorch(...)
for i in range(num_epochs):
    result = est.fit(data, epochs=1)
    model = est.get_model()
    print(f"epoch {i}: {result}")
```

**Comment by jason-dai** [comment]
> Synced offline, here are the brief summary:
> 
> #### Feature Requests:
> Automatically printing loss and other metrics of every epoch or iteration when calling `Estimator.fit`
> 
> #### Current workaround we provides:
> fitting multiple epochs in a loop. E.g.
> 
> **change**
> 
> ```python
> est = Estimator.from_pytorch(...)
> est.fit(data, epochs=epochs)
> ```
> 
> **to**
> 
> ```python
> est = Estimator.from_pytorch(...)
> for i in range(num_epochs):
>     result = est.fit(data, epochs=1)
>     model = est.get_model()
>     print(f"epoch {i}: {result}")
> ```

What should be the desired behavior?

**Comment by yangw1234** [comment]
> 
> What should be the desired behavior?

I think it is a trade off between flexibility and usability.

Our current behavior is more flexible, like pytorch, as it can allow users to checkpointing in their prefered frequency and printing log in their desired format. The downside is they have to write some code. 

If we are tageting usability, I think it also make sense to implement a fixed number of checkpointing and logging stategies and ask user to configure the one closest to their needs.

It is a judgment call.

**Comment by jason-dai** [comment]
> > What should be the desired behavior?
> 
> I think it is a trade off between flexibility and usability.
> 
> Our current behavior is more flexible, like pytorch, as it can allow users to checkpointing in their prefered frequency and printing log in their desired format. The downside is they have to write some code.
> 
> If we are tageting usability, I think it also make sense to implement a fixed number of checkpointing and logging stategies and ask user to configure the one closest to their needs.
> 
> It is a judgment call.

What's the keras behavior?

**Comment by yangw1234** [comment]
> What's the keras behavior?

Keras will automatically output training loss, metrics and speed. Model checkpoint can be configured using model checkpoint callback.
Stick to Keras?

**Comment by jason-dai** [comment]
> > What's the keras behavior?
> 
> Keras will automatically output training loss, metrics and speed. Model checkpoint can be configured using model checkpoint callback. Stick to Keras?

Either Keras or PyTorch Lightening style?

**Comment by jason-dai** [comment]
Do you need to backport to AZ?

**Comment by yangw1234** [comment]
> Do you need to backport to AZ?

I'll backport. This is closed automatically by github.
